sdk/core/Azure.Core/src/OperationOfT.cs

```diff
@@ -43,6 +44,33 @@ public abstract class Operation<T>: Operation where T : notnull
         /// </remarks>
         public abstract ValueTask<Response<T>> WaitForCompletionAsync(CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Periodically calls the server till the long-running operation completes.
+        /// </summary>
+        /// <param name="pollingInterval">
+        /// The interval between status requests to the server.
+        /// The interval can change based on information returned from the server.
+        /// For example, the server might communicate to the client that there is not reason to poll for status change sooner than some time.
+        /// </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the periodical service calls.</param>
+        /// <returns>The last HTTP response received from the server.</returns>
+        /// <remarks>
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final result of the operation.
+        /// </remarks>
+        public virtual Response<T> WaitForCompletion(TimeSpan pollingInterval, CancellationToken cancellationToken)
```

Comment:
This is a good change but some people might be against it. Might be better to have a separate PR for it.

Reply:
is this something that could be added to Azure.Core.Experimental as a way to test and get feedback?

Reply:
Interesting idea. We might be able to ship it as an extension method in experimental.

Reply:
Are there any specific concerns with shipping it in Core?

Reply:
Not from me. Get an API approval early though.

Reply:
We should review it at the arch meeting and once approved change our design guidelines.